src/entities/removeLiquidity/removeLiquidityV3/index.ts

```diff
@@ -0,0 +1,228 @@
+
+                    tokenOutIndex = sortedTokens.findIndex((t) =>
+                        t.isSameAddress(input.amountOut.address),
+                    );
+                }
+                break;
+            case RemoveLiquidityKind.SingleTokenExactIn:
+                {
+                    bptIn = TokenAmount.fromRawAmount(
+                        bptToken,
+                        input.bptIn.rawAmount,
+                    );
+
+                    const { result: minAmountsOut } =
+                        await client.simulateContract({
+                            address: BALANCER_ROUTER[input.chainId],
+                            abi: balancerRouterAbi,
+                            functionName:
+                                'queryRemoveLiquiditySingleTokenExactIn',
+                            args: [
+                                poolState.address,
+                                input.bptIn.rawAmount,
+                                input.tokenOut,
+                                1n, // minAmountOut set to 0 when querying - SC needs it to be > 0 for now - suggested they fix this
+                                '0x',
+                            ],
+                        });
+
+                    amountsOut = sortedTokens.map((t, i) =>
+                        TokenAmount.fromRawAmount(t, minAmountsOut[i]),
+                    );
+                    tokenOutIndex = sortedTokens.findIndex((t) =>
+                        t.isSameAddress(input.tokenOut),
+                    );
+                }
+                break;
+            case RemoveLiquidityKind.Proportional:
+                {
+                    bptIn = TokenAmount.fromRawAmount(
+                        bptToken,
+                        input.bptIn.rawAmount,
+                    );
+
+                    const { result: minAmountsOut } =
+                        await client.simulateContract({
+                            address: BALANCER_ROUTER[input.chainId],
+                            abi: balancerRouterAbi,
+                            functionName: 'queryRemoveLiquidityProportional',
+                            args: [
+                                poolState.address,
+                                input.bptIn.rawAmount,
+                                Array(sortedTokens.length).fill(1n), // minAmountsOut set to 0 when querying
+                                '0x',
+                            ],
+                        });
+                    amountsOut = sortedTokens.map((t, i) =>
+                        TokenAmount.fromRawAmount(t, minAmountsOut[i]),
+                    );
+                    tokenOutIndex = undefined;
+                }
+                break;
+        }
+
+        const output: RemoveLiquidityBaseQueryOutput = {
+            poolType: poolState.type,
+            removeLiquidityKind: input.kind,
+            poolId: poolState.id,
+            bptIn,
```

Comment:
This is showing some intellisense errors for me here and line below.

Reply:
Thanks for pointing that out. It's missing the RemoveLiquidityRecovery implementation. I'll add it over here.

Reply:
Double checked and it's pending SC implementation and Router ABI updates. I'll throw for now and create a task to keep track of it.